packages/medusa/src/services/line-item.ts

```diff
@@ -182,6 +182,101 @@ class LineItemService extends TransactionBaseService {
     )
   }
 
+  /**
+   * Generate a single or multiple line item without persisting the data into the db.
+   *
+   * NOTE: this currently work with IsolatedProductModule flag
+   *
+   * @param variantData
+   * @param context
+   */
+  async generateLineItem<
+    T = LineItemCreateData | LineItemCreateData[],
+    TResult = T extends Array<LineItemCreateData> ? LineItem[] : LineItem
+  >(variantData: T, context: GenerateLineItemContext = {}): Promise<TResult> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        // this.validateGenerateArguments(variantData, context) // TODO: validate for this data shape
```

Comment:
todo:

Reply:
Can we add this now, or will it be included in a follow up?

Reply:
Will add it now